### PR TITLE
update and build against openssl 3

### PR DESCRIPTION
**Pull request by boldorider4**

### CODE REVIEW

**Review by JeanChristopheMorinPerso — COMMENTED** [review]
Can you add a direct runtime dependency on `openssl` please?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ requirements:
     - setuptools-rust
     - setuptools
     - wheel
-    - openssl # [linux]
+    - openssl 3.* # [linux]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - openssl {{ openssl }}  # [linux]
```

**Review by JeanChristopheMorinPerso — DISMISSED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml]: do we need this because we want to build for both `1.1.1` and `3.*`?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Yes, if possible we should build for both.

**Review by lorepirri — COMMENTED** [review]
shall we add and increase the build number?

**Comment by boldorider4** [comment]
> shall we add and increase the build number?

Thanks for catching that!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -34,6 +34,7 @@ requirements:
     - openssl {{ openssl }} # [linux]
   run:
     - python
+    - openssl 3.* # [linux]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    # exact pin handled through openssl run_exports
    - openssl # [linux]
```
We want to build both both OpenSSL 1 and 3.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]
LGTM!

**Pull request merged**